External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -242,7 +264,8 @@
 edgeDistance = round(0.1*averageSize);
 
 % Set the voxels at the border where extreme values were detected to NaN as well
-IM(relativeDelta>thresholdIs & distMap<edgeDistance) = NaN;
+% We update the NaN mask instead of modifying the image
+maskNaN(relativeDelta > thresholdIs & distMapInward < edgeDistance) = 1;
```

Comment:
Tricky, because that means that you have to use the maskNaN throughout. So there is the possibility to take the values that should have been set to NaN. This possibility is not there anymore when they are set to NaNs instead of expanding the NaN mask.

Reply:
There is a maskNaN which says where new values should go. But we also need to know what the "edge or border value" is that we want to extrapolate from. This is not always the edge voxel as in the beginning, it could be that we want to take a few layers deeper, or that we smooth the final few voxels with values for a robust edge value.

Reply:
1. That's what I'm saying - this edge voxels are updated of course based on your exclusion.
After running your code, we update the edge-voxel border on line 296, so we ignore those problematic voxels and extrapolate based on the new edge you have created a few layers deeper.

BTW: Can you indicate - in your example file that you've sent - which are the problematic voxels? I have excluded them based on your code, but I couldnt' really locate any problematic voxels.

Reply:
Then it should not exclude anything right, if there are no problematic voxels? The problematic voxels are there if we e.g. run ExploreASL on low quality and the flowfields are created at [3x3x3] and resampled to [1.5x1.5x1.5]

Reply:
Yes, it should not exclude anything, but it's you who wrote this function ;) With the current dataset you've sent, it excludes something like 1-voxel layer. But I don't see any wrong voxels. So that's why I'm asking if it's really necessary and if we have an example where this helps.

Reply:
I know I wrote this function, which I think should still work but for the exclusion threshold (median+X*MAD), the X might need tweaking? I will send you some other examples.

Reply:
I'm fine with the function in general. I was just saying that I can't see issues, so that's why I can't properly debug and tweak. So no problem with the general functionality.

I'll check your examples. Please test them as well, you might find out that weird values are not so big problem with the new NaN-replacement funciton...